federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -97,8 +100,17 @@ type FederatedInformer interface {
 // framework.DeletionHandlingMetaNamespaceKeyFunc as a keying function.
 type TargetInformerFactory func(*federation_api.Cluster, federation_release_1_4.Interface) (cache.Store, framework.ControllerInterface)
 
+type ClusterLifecycleHandler func(*federation_api.Cluster, FederationView, FederatedReadOnlyStore)
+
+type ClusterLifecycleHandlerFuncs struct {
+	ClusterAvailable   ClusterLifecycleHandler
+	ClusterUnavailable ClusterLifecycleHandler
+}
+
 // Builds a FederatedInformer for the given federation client and factory.
-func NewFederatedInformer(federationClient federation_release_1_4.Interface, targetInformerFactory TargetInformerFactory) FederatedInformer {
+func NewFederatedInformer(federationClient federation_release_1_4.Interface, targetInformerFactory TargetInformerFactory,
+	clusterLifecycle ClusterLifecycleHandlerFuncs) FederatedInformer {
```

Comment:
There are no official K8s go line length guidance. I break line if they feel too long or doesn't fit to my screen. There are other lines that are slightly longer than the one you wanted to break. Anyway I will apply the formatting you asked for.